src/views/forms/FormAccountUnlock/FormAccountUnlockTs.ts

```diff
@@ -82,22 +86,45 @@ export class FormAccountUnlockTs extends Vue {
    * account.
    * @return {void}
    */
+  public isLedger : boolean;
+  checkLedger():boolean {
+    if(this.currentWallet.values.get('type')==WalletType.fromDescriptor("Ledger")){
+      this.isLedger = true
+    } else this.isLedger= false
+    return this.isLedger
+  }
   public processVerification() {
     // - create encrypted payload for active wallet
-    const encrypted = new EncryptedPrivateKey(
+     const encrypted = new EncryptedPrivateKey(
       this.currentWallet.values.get('encPrivate'),
       this.currentWallet.values.get('encIv'),
     )
 
     try {
       // - attempt decryption
       const password = new Password(this.formItems.password)
-      const privateKey: string = encrypted.decrypt(password)
+      const accountService = new AccountService(this.$store)
+      const passwordHash = accountService.getPasswordHash(new Password(this.formItems.password))
+      // read account's password hash and compare
+      const accountPass = this.currentAccount.values.get('password')
 
-      if (privateKey.length === 64) {
-        const unlockedAccount = Account.createFromPrivateKey(privateKey, this.networkType)
-        return this.$emit('success', {account: unlockedAccount, password})
+      if (accountPass !== passwordHash) {
+        return this.$store.dispatch('notification/ADD_ERROR', NotificationType.WRONG_PASSWORD_ERROR)
       }
+        
+      if(this.checkLedger() && accountPass == passwordHash){
```

Comment:
if not currently using a hardware wallet, the password will not be checked anymore ..

Reply:
So when we import a sub wallet in Ledger account by "import an existing account with privateKey", the sub wallet will imported directly without checking with password?.